check/src/substitution.rs

```diff
@@ -294,6 +294,10 @@ impl<T: Substitutable + PartialEq + Clone> Substitution<T> {
     pub fn union(&self, id: &T::Variable, typ: &T) -> Result<(), ()>
         where T::Variable: Clone
     {
+        // Nothing needs to be done if both are the same variable already (also prevents the occurs check from failing)
```

Comment:
No it would fail Typecheck ingredients with `123 occurs in 123` which is obviously nonsense. Unifying with itself should always succeed